internal/testing/test_test.go

```diff
@@ -0,0 +1,94 @@
+package testing
+
+import "testing"
+
+func TestRunTests(t *testing.T) {
+	type args struct {
+		directory   string
+		verbose     bool
+		exitOnError bool
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Non existing directory should return an error",
+			args:    args{directory: "", verbose: false, exitOnError: false},
+			wantErr: true,
+		},
+		{
+			name:    "Directory with no expected.go file should not return an error",
+			args:    args{directory: "testdata/empty_directory", verbose: false, exitOnError: false},
+			wantErr: false,
+		},
+		{
+			name:    "Directory with expected.go file should not return an error",
+			args:    args{directory: "./testdata/accurate_expected", verbose: false, exitOnError: false},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := RunTests(tt.args.directory, tt.args.verbose, tt.args.exitOnError); (err != nil) != tt.wantErr {
+				t.Errorf("RunTests() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_runTest(t *testing.T) {
+	type args struct {
+		directory string
+		verbose   bool
+	}
+	tests := []struct {
+		name    string
+		args    args
```

Comment:
struct not necessary, you always pass `verbose: false`
```suggestion
		directory    string
```